src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -170,134 +167,204 @@ public int updateInternal(Position p) {
         .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
   }
 
+  @InTransaction
+  public int insertPositionEmptyInPerson(String personUuid) {
+    Instant now = Instant.now();
+    return getDbHandle()
+        .createUpdate("/* positionEmptyPerson.insert */ INSERT INTO \"peoplePositions\" "
+            + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
+            + "VALUES (NULL, :personUuid, :createdAt)")
+        .bind("personUuid", personUuid)
+        // Need to ensure this timestamp is greater than previous INSERT.
+        .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+  }
+
+  private int deletePersonEmptyInPositionEmpty() {
+    return getDbHandle()
+        .createUpdate("/* positionEmptyPersonAndPosition.delete */ DELETE FROM \"peoplePositions\" "
+            + " WHERE \"personUuid\" IS NULL and \"positionUuid\" IS NULL")
+        .execute();
+  }
+
+  @InTransaction
+  public int removePersonFromPosition(String positionUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+    numRows += cleanPosition(positionUuid);
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"positionUuid\" = :positionUuid AND \"personUuid\" IS NOT NULL AND "
+            + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + " WHERE \"positionUuid\" = :positionUuid AND \"personUuid\" IS NOT NULL AND "
+                + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).bind("endedAt", DaoUtils.asLocalDateTime(now))
+        .execute();
+
+    if (currentPositionPerson != null) {
+      removePositionFromPerson(currentPositionPerson.getPersonUuid());
+    }
+
+    PersonPositionHistory emptyPersonPosition = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"positionUuid\" = :positionUuid "
+            + " AND \"personUuid\" IS NULL AND \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    if (emptyPersonPosition == null) {
+      numRows += insertPersonEmptyInPosition(positionUuid);
+    }
+
+    numRows += deletePersonEmptyInPositionEmpty();
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
+  }
+
+  @InTransaction
+  public int removePositionFromPerson(String personUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
```

Comment:
Please ensure that each database statement has a unique comment.

Reply:
Fixed with using unique comments.